code/modules/clothing/clothing.dm

```diff
@@ -214,6 +234,12 @@
 		var/mob/M = src.loc
 		M.update_inv_gloves()
 
+/obj/item/clothing/gloves/update_greyscale(list/colors, update)
+	. = ..()
+	if(!greyscale_config)
+		return
+	item_icons = list(slot_gloves_str = icon)
+
```

Comment:
Instead of copy pasting this for every clothing piece couldnt you do it on the parent?